### ISSUE TITLE
Custom Tab: Handle addDefaultShareMenuItem()

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#1401]
User story: https://github.com/mozilla-mobile/reference-browser/issues/209

`CustomTabsIntent.Builder.addDefaultShareMenuItem()`.

**Comment by jonalmeida** [comment]
Looks like we're already handling this in the `CustomTabConfig` as well: https://github.com/mozilla-mobile/android-components/blob/210751ad2b02639cbfc729b0e96a0c2dc204e846/components/browser/session/src/main/java/mozilla/components/browser/session/tab/CustomTabConfig.kt#L109

**Comment by pocmo** [comment]
Yeah, I think `CustomTabConfig` should be pretty much complete for our purposes. However we are not using the data inside it for anyhting yet. :)

**Comment by pocmo** [comment]
> We also need to add a way to find and remove the share menu in the BrowserMenuBuilder when false.

Another option may be to introduce a visible(Session) like function - similar to the toolbar actions - that gets evaluated when building the menu.

**Comment by jonalmeida** [comment]
> I remember we had a UX discussion that lead to the decision to always show the share menu. Maybe that's how we ended up with that default.

I'll stick with it as the default if there was some historical reason for it - a good use case to see how well our APIs hold up for business logic. ;)

> I'm currently working on a PR that will as a side effect add unique ids to menu items. That may be helpful here :)

I was just trying out a solution with a 'tag' as well. I'm glad that I was at least on the right track!



> Another option may be to introduce a visible(Session) like function - similar to the toolbar actions - that gets evaluated when building the menu.

We have a visibile function already, so I can do this as well with pretty trivial logic which should work.
